### PR TITLE
Add cross-origin-embedder-policy tests with service workers

### PR DESCRIPTION
**Pull request by yutakahirano**

### CODE REVIEW

**Comment by yutakahirano** [comment]
@annevk, can you take a look?

@nhiroki, can you take a look at service worker setup code?

**Comment by annevk** [comment]
So all these async functions return a promise resolved with undefined unless something threw an exception. I guess that's fine and not a problem as the only failure condition here is throwing (right?), but it's not really clear to me why this is preferable over not having an async function and simply returning the inner statement. I'll do a more in depth look as well.

File: html/cross-origin-embedder-policy/none-sw-from-none.https.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    reg.installing ? reg.installing : reg.waiting ? reg.waiting : reg.active;
+  await wait_for_state(t, sw, 'activated');
+}, 'setting up');
+
+promise_test(async (t) => {
+  await fetch('resources/nothing-same-origin-corp.txt', {mode: 'no-cors'});
+}, 'making a same-origin request for CORP: same-origin');
+
+promise_test(async (t) => {
+  await fetch('resources/nothing.txt', {mode: 'no-cors'});
```
Comment (annevk) [inline html/cross-origin-embedder-policy/none-sw-from-none.https.html]: Any reason not to use `/common/blank.html`?

**Review by annevk — APPROVED** [review]

File: html/cross-origin-embedder-policy/none-sw-from-none.https.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+function remote(path) {
+  const REMOTE_ORIGIN = get_host_info().HTTPS_REMOTE_ORIGIN;
+  return new URL(path, REMOTE_ORIGIN + '/html/cross-origin-embedder-policy/');
+}
+
+promise_test(async (t) => {
+  const reg = await service_worker_unregister_and_register(t, SCRIPT, SCOPE);
+  const sw =
+    reg.installing ? reg.installing : reg.waiting ? reg.waiting : reg.active;
```
Comment (nhiroki) [inline html/cross-origin-embedder-policy/none-sw-from-none.https.html]: |reg.installing| should always be valid at this point.

File: html/cross-origin-embedder-policy/resources/sw.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+self.addEventListener('activate', event => {
```
Comment (nhiroki) [inline html/cross-origin-embedder-policy/resources/sw.js]: s/event/e/ to make it aligned with the fetch event handler.

File: html/cross-origin-embedder-policy/none-sw-from-none.https.html (hunk 2 above)
Comment (yutakahirano) [reply html/cross-origin-embedder-policy/none-sw-from-none.https.html]: Fixed.

File: html/cross-origin-embedder-policy/none-sw-from-none.https.html [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  await wait_for_state(t, sw, 'activated');
```
[the remark this answers is not in the document]
Comment (yutakahirano) [reply html/cross-origin-embedder-policy/none-sw-from-none.https.html]: Fixed.

File: html/cross-origin-embedder-policy/require-corp-sw-from-none.https.html [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+function remote(path) {
+  const REMOTE_ORIGIN = get_host_info().HTTPS_REMOTE_ORIGIN;
+  return new URL(path, REMOTE_ORIGIN + '/html/cross-origin-embedder-policy/');
+}
+
+promise_test(async (t) => {
+  const reg = await service_worker_unregister_and_register(t, SCRIPT, SCOPE);
+  const sw =
+    reg.installing ? reg.installing : reg.waiting ? reg.waiting : reg.active;
+  await wait_for_state(t, sw, 'activated');
```
[the remark this answers is not in the document]
Comment (yutakahirano) [reply html/cross-origin-embedder-policy/require-corp-sw-from-none.https.html]: Fixed.

File: html/cross-origin-embedder-policy/resources/sw.js (hunk 3 above)
Comment (yutakahirano) [reply html/cross-origin-embedder-policy/resources/sw.js]: Fixed.

File: html/cross-origin-embedder-policy/require-corp-sw-from-require-corp.https.html [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+function remote(path) {
+  const REMOTE_ORIGIN = get_host_info().HTTPS_REMOTE_ORIGIN;
+  return new URL(path, REMOTE_ORIGIN + '/html/cross-origin-embedder-policy/');
+}
+
+promise_test(async (t) => {
+  const reg = await service_worker_unregister_and_register(t, SCRIPT, SCOPE);
+  const sw =
+    reg.installing ? reg.installing : reg.waiting ? reg.waiting : reg.active;
+  await wait_for_state(t, sw, 'activated');
```
[the remark this answers is not in the document]
Comment (yutakahirano) [reply html/cross-origin-embedder-policy/require-corp-sw-from-require-corp.https.html]: Fixed.

File: html/cross-origin-embedder-policy/none-sw-from-require-corp.https.html [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+function remote(path) {
+  const REMOTE_ORIGIN = get_host_info().HTTPS_REMOTE_ORIGIN;
+  return new URL(path, REMOTE_ORIGIN + '/html/cross-origin-embedder-policy/');
+}
+
+promise_test(async (t) => {
+  const reg = await service_worker_unregister_and_register(t, SCRIPT, SCOPE);
+  const sw =
+    reg.installing ? reg.installing : reg.waiting ? reg.waiting : reg.active;
+  await wait_for_state(t, sw, 'activated');
```
[the remark this answers is not in the document]
Comment (yutakahirano) [reply html/cross-origin-embedder-policy/none-sw-from-require-corp.https.html]: Fixed.

File: html/cross-origin-embedder-policy/none-sw-from-none.https.html (hunk 1 above)
Comment (yutakahirano) [reply html/cross-origin-embedder-policy/none-sw-from-none.https.html]: I just forgot about it. Fixed.

**Review by nhiroki — APPROVED** [review]
LGTM

**Comment by yutakahirano** [comment]
Thanks!

**Comment by annevk** [comment]
@yutakahirano you forgot to remove `nothing.txt` itself.

**Comment by yutakahirano** [comment]
Oops, sorry. I will remove it.

**Comment by yutakahirano** [comment]
I uploaded #18586.

**Pull request merged**